File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.js
function solution(board, moves) {
    let answer = 0;
    let dolls = [];
    
    for (let j of moves){
        for (let i = 0; i < board.length; i++){
            let doll = board[i][j-1]
            if (doll !== 0){
                if (dolls && dolls[dolls.length-1] === doll){
                    dolls.pop();
                    answer += 2
                }
                else {dolls.push(doll)}
                board[i][j-1] = 0
                break
            }
        }
    }
    
    return answer;
}

File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.py
def solution(board, moves):
    answer = 0
    dolls = []
    
    for j in moves:
        for i in range(len(board)):
            doll = board[i][j-1]
            if doll != 0:
                if dolls and dolls[-1] == doll:
                    answer += 2
                    dolls.pop()
                else: dolls.append(doll)
                board[i][j-1] = 0
                break

    return answer